infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/external/DefaultHostIngressNamingStrategy.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.server.external;
+
+import io.fabric8.kubernetes.api.model.ServicePort;
+import org.eclipse.che.workspace.infrastructure.kubernetes.Names;
+
+/**
+ * Provides a path-based strategy for exposing service ports outside the cluster using Ingress
+ * Ingresses will be created without an explicit host (defaulting to *).
+ *
+ * <p>This strategy uses different Ingress path entries <br>
+ * Each external server is exposed with a unique path prefix.
+ *
+ * <p>This strategy imposes limitation on user-developed applications. <br>
+ * It should only be used for local development with a single IP address
+ *
+ * <pre>
+ *   Path-Based Ingress exposing service's port:
+ * Ingress
+ * ...
+ * spec:
+ *   rules:
+ *     - http:
+ *         paths:
+ *           - path: service123/webapp        ---->> Service.metadata.name + / + Service.spec.ports[0].name
+ *             backend:
+ *               serviceName: service123      ---->> Service.metadata.name
+ *               servicePort: [8080|web-app]  ---->> Service.spec.ports[0].[port|name]
+ * </pre>
+ *
+ * @author Sergii Leshchenko
+ * @author Guy Daich
+ */
+public class DefaultHostIngressNamingStrategy implements IngressNamingStrategy {
+
+  public static final String DEFAULT_HOST_STRATEGY = "default-host";
+
+  @Override
+  public String getIngressHost(String serviceName, ServicePort servicePort) {
+    return null;
+  }
+
+  @Override
+  public String getIngressName(String serviceName, ServicePort servicePort) {
+    return Names.generateName("ingress");
```

Comment:
I do not really understand why `DefaultHost` should generate name while other strategies return `serviceName + '-' + servicePort.getName();`.
Can not it do in the same way? I mean it even may do `ExternalServerExposer` without extracting this logic to interface, may not it?

Reply:
This logic was present there before my refactoring and I only abstracted it out into this implementation. But you're right - I was a bit intrigued by that as well.

Reply:
The default server strategy should now use the same ingress names as the other two strategies.